include/realizations/catchment/Bmi_Module_Formulation.hpp

```diff
@@ -1113,16 +1113,20 @@ namespace realization {
                 else {
                     provider = forcing.get();
                 }
+
                 // TODO: probably need to actually allow this by default and warn, but have config option to activate
                 //  this type of behavior
                 // TODO: account for arrays later
                 int nbytes = get_bmi_model()->GetVarNbytes(var_name);
                 int varItemSize = get_bmi_model()->GetVarItemsize(var_name);
+                int numItems = nbytes / varItemSize;
+                assert(nbytes % varItemSize == 0);
+
                 std::shared_ptr<void> value_ptr;
                 // Finally, use the value obtained to set the model input
                 std::string type = get_bmi_model()->get_analogous_cxx_type(get_bmi_model()->GetVarType(var_name),
                                                                            varItemSize);
-                if (varItemSize != get_bmi_model()->GetVarNbytes(var_name)) {
+                if (numItems != 1) {
```

Comment:
Okay I wondered this also looking at #534 ... for argument's sake, when are these two conditionals not equivalent? When the model reports the wrong values for one or the other? Or is the new version just clearer?

Reply:
I changed this for clarity and to avoid repetitive calls. It should be functionally equivalent.

Algebra-wise, this is dividing both sides by `varItemSize`. The added `assert` above guards against a bug that makes it not perfectly divisible. I'm trusting that none of these will return negative values. So, that leaves `varItemSize == 0` I guess, which would be equally non-sensical.

Reply:
Oi... yeah there's probably a suite of things like that we _could_ call against BMI models that we should assert true on...e.g. multiplying out get_grid_shape by the get_var_itemsize = get_var_nbytes... something to think about for later, though that would tend to get expensive. The `%` above is pretty cheap though I think.